readme/Readme.scalatex

```diff
@@ -463,31 +559,147 @@
+
+        @sect{BmpParser}
+            @div(id := "bmpdiv")
+            @script(raw("""demo.DemoMain().bmp(document.getElementById("bmpdiv"))"""))
+            @i
+                Given form accepts BMP files and prints short info about it (width, height and etc.).
+                If you haven't any bmp around, you can download classic
+                @a("lena", href:="https://raw.githubusercontent.com/lihaoyi/fastparse/master/byteparse/jvm/src/test/resources/lena.bmp").
+
+            @p
+                @hl.scala{BmpParser} is a good first example of byte parser, it's not small, but structure is pretty simple.
+
+            @i
+                @hl.scala{AnyWordI} and @hl.scala{AnyDwordI} are just parsers that consumes any word (2 bytes) or dword (4 bytes).
+
+            @hl.ref(bytemain/"BmpParser.scala", "val fileHeader")
+
+            @p
+                Bmp file consists of two headers (file header and info header) and pixels in rows with padding,
+                the difficulties are that there are several versions of headers and the parser should distinct them and
+                process them correctly, and that the size and padding of rows in bmp file depends on information from header.
+            @p
+                The first problem is reflected in similar parsers describing 5 versions of info header
+                (@hl.scala{v2HeaderPart}, @hl.scala{v2Header}, @hl.scala{v3HeaderPart}, @hl.scala{v3Header}...),
+                the second in the @hl.scala{bmpRow} function that computes the parameters of row and creates parser on a fly.
+            @p
+                Note also few tricks for parsing binary data.
+            @ul
+                @li
+                    @b
+                        Most of main elements in bmp format has very simple and plain structure. @br
+                     For instance @hl.scala{fileHeader} and @hl.scala{infoHeaderPart}
+                     are just sequences of @hl.scala{AnyDwordI} or @hl.scala{AnyWordI}
+                @li
+                    @b
+                        The extensive usage of @hl.scala{.rep(exactly=...)}.@br
+                    This is due to the fact that many binary formats depends on constant-sized blocks
+                    and the easiest and fastest method to parse them is to write @hl.scala{.rep(exactly=...)}.
+                @li
+                    @b
+                        The same extensive usage of @hl.scala{.flatMap}.@br
+                    @hl.scala{flatMap} allows to retrieve data from one parser and pass it to the next.
+                    The most popular and primitive example is dynamic-sized array with length written at the beginning.
+                    Similar algorithm is used in the @hl.scala{bmp} when header information is passed to the @hl.scala{bmpRow}
+                    that returns new parser for row in this particular bmp file.
+
+        @sect{ClassParser}
```

Comment:
As mentioned in my comment on `Writing Byte Parsers`, let's move this to the new top-level byte-parser section